hive-metastore/src/main/java/org/apache/iceberg/hive/HMSTablePropertyHelper.java

```diff
@@ -0,0 +1,264 @@
+import org.apache.iceberg.view.ViewMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HMSTablePropertyHelper {
+  private static final Logger LOG = LoggerFactory.getLogger(HMSTablePropertyHelper.class);
+  private static final String HIVE_ICEBERG_STORAGE_HANDLER =
+      "org.apache.iceberg.mr.hive.HiveIcebergStorageHandler";
+
+  private static final BiMap<String, String> ICEBERG_TO_HMS_TRANSLATION =
+      ImmutableBiMap.of(
+          // gc.enabled in Iceberg and external.table.purge in Hive are meant to do the same things
+          // but with different names
+          GC_ENABLED, "external.table.purge");
+
+  private HMSTablePropertyHelper() {}
+
+  public static void setHmsTableParameters(
+      String newMetadataLocation,
+      Table tbl,
+      TableMetadata metadata,
+      Set<String> obsoleteProps,
+      boolean hiveEngineEnabled,
+      long maxHiveTablePropertySize,
+      String currentLocation) {
+    Map<String, String> parameters =
+        Optional.ofNullable(tbl.getParameters()).orElseGet(Maps::newHashMap);
+    Map<String, String> summary =
+        Optional.ofNullable(metadata.currentSnapshot())
+            .map(Snapshot::summary)
+            .orElseGet(ImmutableMap::of);
+    // push all Iceberg table properties into HMS
+    metadata.properties().entrySet().stream()
+        .filter(entry -> !entry.getKey().equalsIgnoreCase(HiveCatalog.HMS_TABLE_OWNER))
+        .forEach(
+            entry -> {
+              String key = entry.getKey();
+              // translate key names between Iceberg and HMS where needed
+              String hmsKey = ICEBERG_TO_HMS_TRANSLATION.getOrDefault(key, key);
+              parameters.put(hmsKey, entry.getValue());
+            });
+    setCommonParameters(
+        newMetadataLocation,
+        metadata.uuid(),
+        obsoleteProps,
+        currentLocation,
+        parameters,
+        BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE.toUpperCase(Locale.ENGLISH),
+        metadata.schema(),
+        maxHiveTablePropertySize);
+    setStorageHandler(parameters, hiveEngineEnabled);
+
+    // Set the basic statistics
+    if (summary.get(SnapshotSummary.TOTAL_DATA_FILES_PROP) != null) {
+      parameters.put(StatsSetupConst.NUM_FILES, summary.get(SnapshotSummary.TOTAL_DATA_FILES_PROP));
+    }
+    if (summary.get(SnapshotSummary.TOTAL_RECORDS_PROP) != null) {
+      parameters.put(StatsSetupConst.ROW_COUNT, summary.get(SnapshotSummary.TOTAL_RECORDS_PROP));
+    }
+    if (summary.get(SnapshotSummary.TOTAL_FILE_SIZE_PROP) != null) {
+      parameters.put(StatsSetupConst.TOTAL_SIZE, summary.get(SnapshotSummary.TOTAL_FILE_SIZE_PROP));
+    }
+
+    setSnapshotStats(metadata, parameters, maxHiveTablePropertySize);
+    setPartitionSpec(metadata, parameters, maxHiveTablePropertySize);
+    setSortOrder(metadata, parameters, maxHiveTablePropertySize);
+
+    tbl.setParameters(parameters);
+  }
+
+  public static void setHmsTableParameters(
```

Comment:
I think we should emphasise the difference between the two functions with the same name. Maybe `SetHMSParametersFromTable` and `setHMSParametersFromView`? It's cleaner for future developers to understand the difference.
Also maybe some comments for the functions? Usually the public functions have some explanation for users. One suggestion:
`Based on properties from the Iceberg metadata produces the properties required for HMS. Adds these properties into 'tbl'`

Reply:
I agree with @gaborkaszab that it would be helpful to differentiate the two methods. I'd suggest `setHmsTableParametersForTable` and `setHmsTableParametersForView`.
It would also be helpful to add javadoc for both methods for documentation.

Reply:
The difference between two method is the type, one is Iceberg TableMetadata the other is ViewMetadata. If you pass the first the iceberg table's properties is converted to HMS table, if you pass ViewMetadata then iceberg's view's properties is converted to HMS table. The name comes originally from HiveViewOperations and from HiveTableOperations. I didn't rename them, this is the original naming convention.
If we want to rename let's do it in separate PR and consult with the original author of HiveViewOperations class.

Reply:
No need to consult with anyone who wrote HiveViewOperations. Before this patch we had a `HiveViewOperations.setHmsTableParameters()` and a `HiveTableOperations.setHmsTableParameters()`. It's pretty obvious which one is for a table and which one is for a view. Anyway, both were private functions, so that also reduced the need of documenting the purpose of these functions.
Now, these are public functions, and for public functions in Iceberg people prefer to have at least some documentation about the purpose and the params. Also, I understand that checking the param types could give an indication that one is for tables and the other is for views, but in a public interface I think more clarity is desired and this is something that can be achieved by the naming of the functions.